examples/user_guide/Network_Graphs.ipynb

```diff
@@ -0,0 +1,335 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import holoviews as hv\n",
+    "import networkx as nx\n",
+    "\n",
+    "hv.notebook_extension('bokeh')\n",
+    "\n",
+    "%opts Graph [width=400 height=400]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Visualizing and working with network graphs is a common problem in many different disciplines. HoloViews provides the ability to represent and visualize graphs very simply and easily with facilities for interactively exploring the nodes and edges of the graph, especially using the bokeh plotting interface.\n",
+    "\n",
+    "The ``Graph`` ``Element`` differs from other elements in HoloViews in that it consists of multiple sub-elements. The data of the ``Graph`` element itself are the abstract edges between the nodes, on its own this abstract graph cannot be visualized. In order to visualize it we need to give each node in the ``Graph`` a concrete ``x`` and ``y`` position in form of the ``Nodes``. The abstract edges and concrete node positions are sufficient to render the ``Graph`` by drawing straight-line edges between the nodes. In order to supply explicit edge paths we can also declare ``EdgePaths``, providing explicit coordinates for each edge to follow.\n",
+    "\n",
+    "To summarize a ``Graph`` consists of three different components:\n",
+    "\n",
+    "* The ``Graph`` itself holds the abstract edges stored as a table of node indices.\n",
+    "* The ``Nodes`` hold the concrete ``x`` and ``y`` positions of each node along with a node ``index``. The ``Nodes`` may also define any number of value dimensions, which can be revealed when hovering over the nodes or to color the nodes by.\n",
```

Comment:
Thanks for the reminder, I still have to do some work around this. You will be able to pass in node information without positions in the constructor and it will add the positions automatically. Constructing an abstract ``Nodes`` object without positions doesn't make sense to me though.

Reply:
Done and example added.

Reply:
Ok, sounds good.

Reply:
Just to clarify, is assigning positions to the nodes lazy, in that no positions will be assigned if no one ever tries to visualize it?  If so, that sounds good.

Reply:
> Just to clarify, is assigning positions to the nodes lazy, in that no positions will be assigned if no one ever tries to visualize it?

That sounded good at first but then I realized I don't want to have to reason about a datastructure based on whether it has been looked at or not. In addition, I don't think there would be a reasonable mechanism for such conditional behavior.

Reply:
Hmm.  Automatically assigning node positions for no reasons, for a data structure that could be quite large, also seems unreasonable to me. Not sure what you mean by "conditional", here.

Reply:
Computing a circular layout is not particularly expensive and I don't really see the point of creating a completely abstract graph Element. The way I see it you will instantiate a Graph element when you want to either datashade it or display it directly and both require concrete positions. You also certainly don't want to use HoloViews pickles as a storage format for your **huge** graph. Finally, we don't have the concept for a conditionally displayable element anywhere else, so overall I think it's the most sensible behavior, unless you have a concrete use case for actually creating a Graph with raw edges.

Reply:
Computing a circular layout isn't expensive in time, but it is expensive in storage, at least for a sparse graph -- each node then gets two extra columns of data.  I'm still not sure what either of you mean by "conditional"; conditional on what?  I was suggesting that the element always be displayable, just that this extra data not be computed until precisely the point where it first needs to be displayed.

The use case I imagine for that is to declare a huge Graph but then use HoloViews machinery to select only a subset of it to make a much smaller Graph that you actually display. I'm not worried about pickling it, just about making the handy tools in HoloViews available without forcing the Graph to be something displayable, as that's a separate consideration.

Reply:
>I was suggesting that the element always be displayable, just that this extra data not be computed until precisely the point where it first needs to be displayed.

Okay sure, I can make it a property which computes nodes the first time they are accessed, which will basically happen when you call any method.

Reply:
@jlstevens, is that ok?  If you veto it, I won't object; up to you.

Reply:
If the idea of the property is simply to defer assigning positions in a way that you can't tell the positions weren't there all along (except by tracking memory usage) then that seems ok to me. I won't veto it and I'll let Philipp decide although it does strike me a little bit as being premature optimization for a case we don't have a real example of yet.

Reply:
Pushed it now, revert it if you do object.